src/generate.rs

```diff
@@ -381,36 +435,31 @@ fn register_block(registers: &[Register], defs: &Defaults) -> Result<Tokens> {
             );
             i += 1;
         }
-
-        let comment = &format!(
-            "0x{:02x} - {}",
-            register.offset,
-            util::respace(&register.info.description)
-        )
-                           [..];
-
-        let rty = match register.ty {
-            Either::Left(ref ty) => Ident::from(&**ty),
-            Either::Right(ref ty) => Ident::from(&***ty),
+        
+        fields.push( util::register_to_rust(&register) );
+        
+        offset = match register {
+            Register::Single(ref _into) => register.address_offset
+                +
+                register
+                .size
+                .or(defs.size)
+                .ok_or_else(
+                    || {
+                        format!("Register {} has no `size` field", register.name)
+                    },
+                )? / 8,
+            Register::Array(ref _into, ref array_info) => register.address_offset
+                +
+                register
+                .size
+                .or(defs.size)
+                .ok_or_else(
+                    || {
+                        format!("Register {} has no `size` field", register.name)
+                    },
+                )? * array_info.dim / 8,
```

Comment:
nit: I know that it's unrelated to this PR but could you replace the `8` by a `BITS_PER_BYTE` constant?